src/reflect/scala/reflect/internal/tpe/TypeConstraints.scala

```diff
@@ -127,18 +127,12 @@ private[internal] trait TypeConstraints {
         case _            => !lobounds.contains(tp)
       }
       if (mustConsider) {
-        def justTwoStrings: Boolean = (
-          tp.typeSymbol == StringClass && tp.isInstanceOf[ConstantType] &&
-          lobounds.lengthCompare(1) == 0 && lobounds.head.typeSymbol == StringClass
-        )
         if (isNumericBound && isNumericValueType(tp)) {
           if (numlo == NoType || isNumericSubType(numlo, tp))
             numlo = tp
           else if (!isNumericSubType(tp, numlo))
             numlo = numericLoBound
         }
-        else if (justTwoStrings)
-          lobounds = tp.widen :: Nil // don't accumulate strings; we know they are not exactly the same bc mustConsider
```

Comment:
Not quite - `mustConsider` is a necessary but not sufficient condition. I mean that `!mustConsider` implies they are the same, but `mustConsider` doesn't imply they are not the same. `!lobounds.contains(tp)` is for optimization, `!lobounds.exists(_ =:= tp)` would be for semantics, but you probably don't want to that here.

`$dealiasWidenWarning` gives some hints:

> Note that type aliases can hide beneath singleton types and singleton types can hide inside type aliases. Moreover, aliases might lurk in the upper bounds of abstract types. Therefore careful thought has to be applied to identify and carry out unwrapping logic specific to your use case.

Or in this case passing a different instance of the same type would be enough.

Reply:
If there were an "ears" reaction emoji, I would have used it here.

Reply:
worth adding this was a last-second tweak, so not surprising that I didn't see to the bottom of the pond, or in this case, the Caribbean Sea.